### ISSUE TITLE
1.0.2 need to save SASS file two times in order to get rid of errors

### ISSUE THREAD

**Issue by weilinzung** [webpack-contrib/stylelint-webpack-plugin#194]
- Operating System: macOS Mojave 10.14.5
- Node Version: v11.13.0
- NPM Version: 6.10.0
- webpack Version: 4.41.2
- stylelint-webpack-plugin Version: 1.0.2

### Expected Behavior
I am using `webpack-dev-serve`, `new webpack.NoEmitOnErrorsPlugin()` & `sass-loader` with this plugin, on`v0.10.5` everything works fine.
I just save one time of the file after I fix the SASS that errors would be gone.

### Actual Behavior
When I update to `v1.0.2`, every time when an SCSS file has errors, I have to save the file 2 times in order to make the error go away. Basically, the error logs are duplicated for the first time and the second time. 

first time log  from -> sass-loder
```
SassError: Invalid CSS after "}": expected 1 selector or at-rule, was "html"
        on line 4 of src/_scss/_general.scss
```

second time log from -> stylelint-webpack-plugin
```
src/_scss/_general.scss
 7:3  ✖  Unknown word   CssSyntaxError
```

### Code

Here is my setup:

```js
    new StyleLintPlugin({
      failOnError: false,
      emitErrors: true,
      configFile: `stylelintrc.js`,
      syntax: 'scss',
      files: '*.s?(a|c)ss'
    }),
```

### How Do We Reproduce?

Run `stylelint-webpack-plugin` using `webpack-dev-server` in watch mode.

Question, what changes on `failOnError` & ` emitErrors` from the last two releases?

Thank you!

**Comment by weilinzung** [comment]
Update: once I remove `new webpack.NoEmitOnErrorsPlugin()` with the `v1.0.2` would be ok, but I think I need the `new webpack.NoEmitOnErrorsPlugin()`.

**Comment by weilinzung** [comment]
Test again when I change back this line back to the old version that would work: https://github.com/webpack-contrib/stylelint-webpack-plugin/compare/v1.0.0...master#diff-195523ef24b1bd918b6f5644d00b5b55L24

From
`compiler.hooks.afterEmit.tapAsync(`
to
`
compiler.hooks.afterCompile.tapAsync(
`

**Comment by ricardogobbosouza** [comment]
@weilinzung could provide a repository for reproduction?

**Comment by weilinzung** [comment]
@ricardogobbosouza sorry, it is an internal project that I don't have it in the public. Just wondering why to change the hooks to `afterCompile`.

**Comment by ricardogobbosouza** [comment]
@weilinzung I understand, if you can provide a demo repository with this error, I appreciate.
The `afterCompile` caused problems with the `lintDirtyModulesOnly` option, I'm analyzing this change

**Comment by weilinzung** [comment]
@ricardogobbosouza Sorry for the delay, seems you fix this. When would this release?

**Comment by ricardogobbosouza** [comment]
@weilinzung I will release soon

**Comment by ricardogobbosouza** [comment]
@weilinzung released v1.0.4

**Comment by weilinzung** [comment]
@ricardogobbosouza amazing! I would test now!

**Comment by weilinzung** [comment]
@ricardogobbosouza I think this issue comes back again with `v1.1.2`.

**Comment by ricardogobbosouza** [comment]
Hi @weilinzung 
The hook correct is `afterEmit` and not `afterCompile`

**Comment by ricardogobbosouza** [comment]
@weilinzung could provide a repository for reproduction?

**Comment by weilinzung** [comment]
@ricardogobbosouza I still not have it, but here is my setup.

I try something else by removing the `new webpack.NoEmitOnErrorsPlugin()`, then would work again with the latest `1.1.2`. I also have `sass-loader` that would report syntax errors.

```
    // stylelint
    new StyleLintPlugin({
      failOnError: false,
      emitErrors: true,
      configFile: `stylelintrc.js`,
      syntax: 'scss',
      files: './src/**/*.s?(a|c)ss'
    }),

    // hot module replacement
    new webpack.HotModuleReplacementPlugin(),
    // not reload if there is compile error
    new webpack.NoEmitOnErrorsPlugin()
```

**Comment by ricardogobbosouza** [comment]
Try this @weilinzung 

```js
new StylelintPlugin({
  emitWarning: true,
  configFile: `stylelintrc.js`,
  syntax: 'scss',
  files: './src/**/*.s?(a|c)ss'
}),
....
```

Use `emitWarning` see https://github.com/webpack-contrib/stylelint-webpack-plugin/issues/200#issuecomment-570824384

**Comment by weilinzung** [comment]
@ricardogobbosouza thank you, but that does not work for me, but remove option plugin`new webpack.NoEmitOnErrorsPlugin()` it is fine for now.
